airflow/providers/amazon/aws/triggers/ecs.py

```diff
@@ -179,8 +181,7 @@ async def run(self) -> AsyncIterator[TriggerEvent]:
                 finally:
                     if self.log_group and self.log_stream:
                         logs_token = await self._forward_logs(logs_client, logs_token)
-
-        yield TriggerEvent({"status": "success", "task_arn": self.task_arn})
+        raise AirflowException("Waiter error: max attempts reached")
```

Comment:
This should yield a failure event instead

Reply:
I disagree, yielding an event will send this back to the scheduler, and require a worker to be setup and everything just to throw an error there. Might as well save all this by raising directly in the triggerer.

I was discussed quickly in this slack thread: https://apache-airflow.slack.com/archives/CCPRP7943/p1685663968275419

Reply:
Sound good @uranusjr ?